### PR TITLE
[RHELC-1578] Change out of date packages overridable inhibitor to warning

### PR DESCRIPTION
**Pull request by pr-watson**
This PR changes the overridable inhibitor result `OUT_OF_DATE_PACKAGES` to a warning. Relevant unit tests have been updated.

Jira Issues:

- [RHELC-1578](https://issues.redhat.com/browse/RHELC-1578)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` or `[HMS-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: convert2rhel/breadcrumbs.py [hunk 1]
```diff
@@ -74,6 +74,8 @@ def __init__(self):
         self.run_id = "null"
         # The convert2rhel package object from the yum/dnf python API for further information extraction.
         self._pkg_object = None
+        #
+        self.non_default_channel = "null"
```
Comment (r0x0d) [inline convert2rhel/breadcrumbs.py:78]: Is this related to the PR?

Comment (pr-watson) [reply convert2rhel/breadcrumbs.py:78]: Sorry that was from another

Comment (r0x0d) [reply convert2rhel/breadcrumbs.py]: alrighty

File: convert2rhel/unit_tests/actions/system_checks/package_updates_test.py [hunk 2]
```diff
@@ -83,61 +83,20 @@ def test_check_package_updates_not_up_to_date(pretend_os, monkeypatch, package_u
...
+                id="OUT_OF_DATE_PACKAGES",
                 title="Outdated packages detected",
                 description="Please refer to the diagnosis for further information",
                 diagnosis=diagnosis,
                 remediations="Run yum update to update all the packages on the system.",
+                variables={},
             ),
         )
     )
-    package_updates_action.run()
```
Comment (r0x0d) [inline convert2rhel/unit_tests/actions/system_checks/package_updates_test.py:140]: I believe this was removed by mistake?

Comment (pr-watson) [reply convert2rhel/unit_tests/actions/system_checks/package_updates_test.py:140]: The formatting just looks like that because I modified the test from being an assert action result to asserting an action message set. The run call is there present - https://github.com/oamg/convert2rhel/blob/b8d3c5734561bf0ea4333215d516db2034b4752c/convert2rhel/unit_tests/actions/system_checks/package_updates_test.py#L127

File: convert2rhel/actions/system_checks/package_updates.py [hunk 3]
```diff
@@ -79,48 +77,23 @@ def run(self):
...
-                title="Skip package not up to date check",
-                description=skip_package_not_up_to_date_message,
-            )
-
             logger.warning(package_not_up_to_date_error_message)
             self.add_message(
                 level="WARNING",
-                id="OUTDATED_PACKAGE_MESSAGE",
+                id="OUT_OF_DATE_PACKAGES",
                 title="Outdated packages detected",
```
[the remark this answers is not in the document]
Comment (pr-watson) [reply convert2rhel/actions/system_checks/package_updates.py:91]: This message was output if the `CONVERT2RHEL_OUTDATED_PACKAGE_CHECK_SKIP` env var was used to skip the OVERRIDABLE inhibitor that was in place. Since we have changed the OVERRIDABLE inhibitor to a warning now, we no longer need `OUTDATED_PACKAGE_MESSAGE`. `OUT_OF_DATE_PACKAGES` isn't a new ID, it is just formatted liked this because the lines from `OUTDATED_PACKAGE_MESSAGE` match what was there for the `OUT_OF_DATE_PACKAGES` result. This is how the final formatting looks - https://github.com/oamg/convert2rhel/blob/b8d3c5734561bf0ea4333215d516db2034b4752c/convert2rhel/actions/system_checks/package_updates.py#L9

File: config/convert2rhel.ini [hunk 4]
```diff
@@ -19,7 +19,6 @@
 [inhibitor_overrides]
 # incomplete_rollback              = false
 # tainted_kernel_module_check_skip = false
-# outdated_package_check_skip      = false
```
Comment (hosekadam) [inline config/convert2rhel.ini:22]: Probably I would also remove the `outdated_package_check_skip` from the `/toolopts/config.py` (mapping and `FileConfig`). And depending what get merged first also from [`/utils/__init__.py` (`warn_deprecated_env()`)](https://github.com/oamg/convert2rhel/pull/1387/files#diff-a180a3ef19514b4d0fe0a42f5d5e60abdfdce708992bcd2363de752aeaf30561R1128)

File: convert2rhel/actions/system_checks/package_updates.py [hunk 5]
```diff
@@ -79,48 +77,23 @@ def run(self):
...
                 description="Please refer to the diagnosis for further information",
                 diagnosis=package_not_up_to_date_error_message,
                 remediations="Run yum update to update all the packages on the system.",
             )
+            return
```
Comment (hosekadam) [inline convert2rhel/actions/system_checks/package_updates.py:96]: I would say the `return` or the following `else` is not needed.

Comment (r0x0d) [reply convert2rhel/actions/system_checks/package_updates.py:96]: I would vote for removing the else and keeping the return

Comment (pr-watson) [reply convert2rhel/actions/system_checks/package_updates.py]: Alright I'll remove the else

**Review by Venefilyn — APPROVED** [review]

**Review by Andrew-ang9 — COMMENTED** [review]
This looks good to me, the integration tests are passing and I think the ones that are failing are the ones we currently expect to fail. So once the conflict is taken care of should be good to go!

**Comment by pr-watson** [comment]
Sounds good let me fix this conflict

**Comment by pr-watson** [comment]
@danmyway looks like these integration test failures are not related to the PR, can this be merged?

**Review by danmyway — APPROVED** [review]
The test failures are indeed unrelated, can be merged.
